### PR TITLE
type_switch(): Simplify more code

### PR DESCRIPTION
**Pull request by GindaChen**
Use `type_switch()` to simplify code. 

**Known issues**
- `__FUNCTION__` macro used in lambda only outputs `operator()` because invoking lambda function is equivalent to invoking a structure's operator(). We may need to switch to find some extern lib.
- Performance _may_ degrade if we first loop on rows and then columns.

### CODE REVIEW

File: src/storage/table.cc [hunk 1]
```diff
@@ -114,55 +115,33 @@ void DBTable::InsertRecords(
...
   auto block = GetBlockForInsert();
   int offset = 0;
-  int length = l;
-  // optimize?
-  int column_types[num_cols];
-  for (int i = 0; i < num_cols; i++) {
-    column_types[i] = schema->field(i)->type()->id();
-  }
+
+  // TODO: (Optimize) Why don't we switch column type then loop over rows?
```
Comment (GindaChen) [inline src/storage/table.cc:119]: @srsuryadev Why don't we switch column type then loop over rows? If the lambda function is not inlined, we may pay a huge cost here.

File: src/operators/select/select.cc [hunk 2]
```diff
@@ -126,159 +127,105 @@ arrow::Datum Select::Filter(const std::shared_ptr<Block> &block,
...
+  // TODO: There are a lot of assumptions over the type T.
+  auto numeric_filter = [&]<typename T>(T *) -> arrow::Datum {
+    using CType = ArrowGetCType<T>;
+    using ScalarType = ArrowGetScalarType<T>;
+
+    CType val =
+        std::static_pointer_cast<ScalarType>(predicate->value_.scalar())->value;
+    auto datum_val = predicate->value_;
+
+    switch (predicate->comparator_) {
```
Comment (srsuryadev) [inline src/operators/select/select.cc:160]: I feel this switch can be removed with help of a map (comp->primitive comp) or any other means like type switch, maybe we can handle in another PR

File: src/storage/table.cc (hunk 1 above)
Comment (srsuryadev) [reply src/storage/table.cc:119]: I don't see any reason, it can be modified to our needs!

File: src/operators/aggregate/hash_aggregate.cc [hunk 3]
```diff
@@ -569,105 +569,104 @@ void HashAggregate::Finish(Task *ctx) {
...
+      using BuilderType = ArrowGetBuilderType<T>;
+
+      auto builder =
+          std::reinterpret_pointer_cast<BuilderType>(getBuilder(data_type));
+
+      for (auto const it : *global_tuple_map) {
+        int chunk_id, item_index;
+        auto hash_key = it.first;
+        std::tie(chunk_id, item_index) = it.second;
+        auto raw_chunk = group_by.chunks().at(chunk_id);
```
Comment (srsuryadev) [inline src/operators/aggregate/hash_aggregate.cc:598]: I feel we can avoid auto as much as possible for better readability [Optional]

File: src/storage/table.cc (hunk 1 above)
Comment (srsuryadev) [reply src/storage/table.cc:119]: I don't see any reason, it can be modified to our needs!

**Review by srsuryadev — APPROVED** [review]
LGTM! (Few suggestions)

File: src/operators/aggregate/hash_aggregate.cc (hunk 3 above)
Comment (pateljm) [reply src/operators/aggregate/hash_aggregate.cc:598]: That is a good point -- excessive use of auto makes code hard to write. Adding comments to clarify the type can help, but if we don't need auto, we should avoid it.

File: src/storage/table.cc (hunk 1 above)
Comment (GindaChen) [reply src/storage/table.cc:119]: We can open another issue to address this. I don't have a good idea to break this in this PR.

File: src/operators/aggregate/hash_aggregate.cc (hunk 3 above)
Comment (GindaChen) [reply src/operators/aggregate/hash_aggregate.cc:598]: Put this into the todo list in #58, when we refactor hash aggregate.

File: src/operators/select/select.cc (hunk 2 above)
Comment (GindaChen) [reply src/operators/select/select.cc]: Done with a comparator switcherr.

**Pull request merged**